APMrover2/mode_auto.cpp

```diff
@@ -151,6 +177,50 @@ void ModeAuto::start_RTL()
     }
 }
 
+// hand over control to external navigation controller in AUTO mode
+void ModeAuto::start_guided(const Location& loc)
+{
+    if (rover.mode_guided.enter()) {
+        _submode = Auto_Guided;
+
+        // initialise guided start time and position as reference for limit checking
+        rover.mode_guided.limit_init_time_and_location();
+
+        // sanity check target location
+        Location loc_sanitised = loc;
+        if ((loc_sanitised.lat != 0) || (loc_sanitised.lng != 0)) {
+            location_sanitize(rover.current_loc, loc_sanitised);
+            guided_target = loc_sanitised;
+            guided_target_valid = true;
+        } else {
+            guided_target_valid = false;
+        }
+    }
+}
+
+// send latest position target to offboard navigation system
+void ModeAuto::send_guided_position_target()
+{
+    if (!guided_target_valid) {
+        return;
+    }
+
+    // send at maximum of 1hz
+    uint32_t now_ms = AP_HAL::millis();
+    if ((guided_target_sent_ms == 0) || (now_ms - guided_target_sent_ms > AUTO_GUIDED_SEND_TARGET_MS)) {
+        guided_target_sent_ms = now_ms;
+
+        // get system id and component id of offboard navigation system
+        uint8_t sysid = 0;
+        uint8_t compid = 0;
+        mavlink_channel_t chan;
+        if (GCS_MAVLINK::find_by_mavtype(MAV_TYPE_ONBOARD_CONTROLLER, sysid, compid, chan)) {
+            gcs().chan(chan-MAVLINK_COMM_0).send_set_position_target_global_int(sysid, compid, guided_target);
```

Comment:
`find_by_mavtype` should return a GCS_MAVLINK* rather than a chan.  Not your fight in this PR, of course.